plugins/file/plugin.go

```diff
@@ -21,16 +26,74 @@ func init() {
 	plugins.RegisterPlugin("file", setupFile6, setupFile4)
 }
 
+//Record holds an IP lease record
+type Record struct {
+	IP        net.IP
+	leaseTime uint32
+	leased    int64
+}
+
 // StaticRecords holds a MAC -> IP address mapping
 var StaticRecords map[string]net.IP
 
 // DHCPv6Records and DHCPv4Records are mappings between MAC addresses in
 // form of a string, to network configurations.
 var (
 	DHCPv6Records map[string]net.IP
-	DHCPv4Records map[string]net.IP
+	DHCPv4Records map[string]Record
+	LeaseTime     uint32
+	filename      string
+	serverIP      net.IP
+	netmask       net.IP
```

Comment:
Why not `net.IPMask` here?

Reply:
`net.IPMask ` does not have the functions necessary in `CreateIP` functions

Reply:
> `net.IPMask` does not have the functions necessary in `CreateIP` functions

Which functions are those?

Reply:
Nevermind i fixed it.